homeassistant/helpers/event.py

```diff
@@ -370,7 +370,9 @@ def pattern_time_change_listener(event):
         last_now = now
 
         if next_time <= now:
-            hass.async_run_job(action, event.data[ATTR_NOW])
+            if local:
+                now = dt_util.as_local(now)
```

Comment:
Does it matter that we now pass local time to `calculate_next`?

Reply:
https://github.com/home-assistant/home-assistant/pull/21630